lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.cpp

```diff
@@ -699,6 +702,159 @@ void ProxyRenderDelegate::_UpdateSelectionStates()
     }
 }
 
+void ProxyRenderDelegate::_UpdateRenderTags()
+{
+    // USD pulls the required render tags from the task list passed into execute.
+    // Only rprims which are dirty & which match the current set of render tags
+    // will get a Sync call.
+    // Render tags are harder for us to handle than HdSt because we have our own
+    // cached version of the scene in MPxSubSceneOverride. HdSt draws using
+    // HdRenderIndex::GetDrawItems(), and that returns only items that pass the
+    // render tag filter. There is no need for HdSt to do any update on items that
+    // are being hidden, because the render pass level filtering will prevent
+    // them from drawing.
+    // The Vp2RenderDelegate implements render tags using MRenderItem::Enable(),
+    // which means we do need to update individual MRenderItems when the render
+    // tags change.
+    // When we change the desired render tags on the proxyShape we'll be adding
+    // and/or removing some tags, so we can have existing MRenderItems that need
+    // to be hidden, or hidden items that need to be shown.
+    // This function needs to do three things:
+    //  1: Make sure every rprim with a render tag whose visibility changed gets
+    //     marked dirty. This will ensure the upcoming execute call will update
+    //     the visibility of the MRenderItems in MPxSubSceneOverride.
+    //  2: Make a special call to Execute() with only the render tags which have
+    //     been removed. We need this extra call to hide the previously shown
+    //     items.
+    //  3: Update the render tags with the new final render tags so that the next
+    //     real call to execute will update all the now visible items.
+    bool renderPurposeChanged, proxyPurposeChanged, guidePurposeChanged;
+    _proxyShapeData->UpdatePurpose(
+        &renderPurposeChanged, &proxyPurposeChanged, &guidePurposeChanged);
+    if (renderPurposeChanged || proxyPurposeChanged || guidePurposeChanged)
+    {
+        // Build the list of render tags which were added or removed (changed)
+        // and the list of render tags which were removed.
+        TfTokenVector changedRenderTags;
```

Comment:
Let's add profiler instrumentation to this method. This will allow us to keep an eye on performance and scalability.